### PR TITLE
Implement Edit Class Command

### LINKED ISSUE #46: Change a class name
As a TA, I can change a class name so that the name of my class remains up to date

### LINKED ISSUE #47: Update a class note
As a TA, I can update a class note so that I can keep track of pointers to look out for in a particular class

### PR DESCRIPTION
**Pull request by songgthu**
Functionality: 
- Edit class name (/n), class memo (/m), class schedule (/t)
- Order of parameters does not matter
- Can edit 1 or more parameters at same time (e.g: `edit /c 1 /m submit marking report /n cs2100`)

New classes:
- EditClassCommand
- EditClassCommandParser
- Memo
- Schedule: follows the format: weekDay, HH:MM-HH:MM (e.g: mon, 15:00-17:00)

Other changes:
- Update testcases to match the Class constructor for Memo, Schedule
- Add new testcases to reach the code coverage target
- Remove TempClass class in student package

### CODE REVIEW

**Review by sudarshan2401 — APPROVED** [review]
LGTM! The implementations of EditClassCommand, Memo and Schedule are really good and easy to understand!

File: docs/UserGuide.md [hunk 1]
```diff
@@ -59,13 +59,16 @@ Coming soon...
 
 * If you are using a PDF version of this document, be careful when copying and pasting commands that span multiple lines as space characters surrounding line-breaks may be omitted when copied over to the application.
   </box>
+<<<<<<< HEAD
```
Comment (seanpzk) [inline docs/UserGuide.md]: Thanks for editing the UG! I think you missed out on this merge conflict message hahaha

**Review by DonovanJJ — APPROVED** [review]
LGTM! 
Very well documented code and easy to follow 👍

**Review by seanpzk — DISMISSED** [review]
LGTM! Might want to clean up the merge conflict messages 😄

**Pull request merged**